src/FunderMaps.WebApi/Controllers/Report/InquiryController.cs

```diff
@@ -22,184 +20,264 @@ namespace FunderMaps.WebApi.Controllers.Report
+                contentType: input.ContentType,
+                stream: input.OpenReadStream());
 
             var output = new DocumentDto
             {
-                Name = fileName,
+                Name = storeFileName,
             };
 
             // Return.
             return Ok(output);
         }
 
+        // GET: api/inquiry/download
+        /// <summary>
+        ///     Retrieve document access link.
+        /// </summary>
+        [HttpGet("{id:int}/download")]
+        public async Task<IActionResult> GetDocumentAccessLinkAsync(int id)
+        {
+            // Act.
+            var inquiry = await _inquiryRepository.GetByIdAsync(id);
+            var link = await _blobStorageService.GetAccessLinkAsync(
+                containerName: Core.Constants.InquiryStorageFolderName,
+                fileName: inquiry.DocumentFile,
+                hoursValid: 1);
+
+            // Map.
+            var result = new BlobAccessLinkDto
+            {
+                AccessLink = link
+            };
+
+            // Return.
+            return Ok(result);
+        }
+
+        // PUT: api/inquiry/{id}
+        /// <summary>
+        ///     Update inquiry by id.
+        /// </summary>
         [HttpPut("{id:int}")]
         public async Task<IActionResult> UpdateAsync(int id, [FromBody] InquiryDto input)
         {
             // Map.
-            var inquiry = _mapper.Map<Inquiry>(input);
+            var inquiry = _mapper.Map<InquiryFull>(input);
             inquiry.Id = id;
 
             // Act.
-            await _inquiryUseCase.UpdateAsync(inquiry);
+            await _inquiryRepository.UpdateAsync(inquiry);
+
+            // FUTURE: Does this make sense?
+            // Only when this item was rejected can we move into
+            // a pending state after update.
+            if (inquiry.State.AuditStatus == AuditStatus.Rejected)
+            {
+                // Transition.
+                inquiry.State.TransitionToPending();
+
+                // Act.
+                await _inquiryRepository.SetAuditStatusAsync(inquiry.Id, inquiry);
+            }
 
             // Return.
             return NoContent();
         }
 
+        // POST: api/inquiry/{id}/status_review
+        /// <summary>
+        ///     Set inquiry status to review by id.
+        /// </summary>
         [HttpPost("{id:int}/status_review")]
         public async Task<IActionResult> SetStatusReviewAsync(int id, StatusChangeDto input)
         {
             // Act.
-            await _inquiryUseCase.UpdateStatusAsync(id, AuditStatus.PendingReview, input.Message);
+            var inquiry = await _inquiryRepository.GetByIdAsync(id);
+
+            // Transition.
+            inquiry.State.TransitionToReview();
+
+            // Act.
+            await _inquiryRepository.SetAuditStatusAsync(inquiry.Id, inquiry);
+            await _notificationService.NotifyByEmailAsync(
+                address: new string[] { "info@example.com" }, // TODO:
```

Comment:
Don't we want this non-hardcoded for a master merge?

Reply:
True, and we do have a configuration option for this value. However the structure is not yet final

Reply:
Check